src/SprykerSdk/Zed/ComposerConstrainer/Business/ComposerConstrainerFacadeInterface.php

```diff
@@ -24,4 +24,11 @@ public function updateConstraints(): ComposerConstraintCollectionTransfer;
      * @return \Generated\Shared\Transfer\ComposerConstraintCollectionTransfer
      */
     public function validateConstraints(): ComposerConstraintCollectionTransfer;
+
+    /**
+     * @api
+     *
+     * @return \Generated\Shared\Transfer\ComposerConstraintCollectionTransfer
+     */
+    public function validateCoreAndForeignConstraints(): ComposerConstraintCollectionTransfer;
```

Comment:
On the second thought, do we need to run this command on core classes with this flag?
Not it looks like violation of SRP at some point

Reply:
As for my opinion it is the question of BL. The only reason we have two options here is the BC. In fact, I don't see any reasoning for skipping anything from the validation and update. Thus if it was the initial release of the ComposerConstrainer - it might be just check all dependancies without any filtering.